sharc/propagation/propagation_factory.py

```diff
@@ -68,6 +69,16 @@ def create_propagation(
             return PropagationSatSimple(random_number_gen)
         elif channel_model == "TerrestrialSimple":
             return PropagationTerSimple(random_number_gen)
+        elif channel_model == "P528":
+            # Check IF ther's P528 parameters
+            if hasattr(param_system, "param_p528"):
+                return PropagationP528(
+                    random_number_gen=random_number_gen,
+                    time_percentage=param_system.param_p528.time_percentage,
+                    polarization=param_system.param_p528.polarization,
+                )
+            # Use defaults if no parameters specified
+            return PropagationP528(random_number_gen=random_number_gen, time_percentage="RANDOM", polarization="RANDOM")
```

Comment:
Defaults are already set on `ParametersP528`. Using a default parameter here may lead to faulty behavior. For example, a parameter file may run using `channel_model = "P528"`, and setting `param_p528.polarization = 1`. However, if `param_p528` is not created inside `param_system`, the default values will be used.

It is better to simply error when the system does not have an attribute named `param_p528` than to silently use default values.